pytket_qir/qir.py

```diff
@@ -228,6 +240,25 @@ def block_to_circuit(self, block: QirBlock, circuit: Circuit) -> Circuit:
                 circuit.add_gate(op, unitids)
             elif instr.instr.is_qir_call:  # Setting the conditional bit for branching.
                 c_reg_index = instr.instr.call_func_params[0].constant.result_static_id
+            elif instr.instr.is_call:  # WASM external call.
+                if not self.wasm_handler:
+                    raise ValueError("A WASM file handler must be provided.")
+                func_name = instr.func_name
+                assert func_name
+                matched_str = re.search("__quantum__(.+?)__(.+?)__(.+)", func_name)
+                assert matched_str
```

Comment:
I think these asserts should be exceptions, as the conditions are outside the control of this module?

Reply:
OK, got that. More generally, do you mean that if conditions are external to the module for it to possibly recover, then that should be an exception ?

Reply:
Done.